### ISSUE TITLE
Use Telegram for agent communication / notifications

### ISSUE THREAD

**Issue by vizion8-dan** [zammad/zammad#793]
### Infos:

* Used Zammad version: 1.3.x
* Operating system: Linux
* Browser + version: 

### Expected behavior:

* Feature request: Telegram Bot/channel integration could be used as an alternate way for agent communication  (for notifications, escalations).

### Actual behavior:

* Agents are notified by email or browser notifications.

**Comment by nublaii** [comment]
It would be great if agents could set a telegram username (@alias) on their profile and add a notification option on triggers for telegram->owner.

**Comment by gituser** [comment]
Any update on this issue?

I see there is a telegram bot integration implemented so should be very easy to implement this feature to notify about new / pending / etc tickets similar as there is a Slack notification implemented.

EDIT:

For those who are looking to resolve this: google for zammad triggers - https://admin-docs.zammad.org/en/latest/manage/trigger.html

Basically, you create a trigger with custom hook notification, paste there an URL with your script.

Script needs to handle `Content-type: application/json` as zammad will send this content with POST method to your URL.

The script should then handle the info and post to the telegram.

You can also create multiple triggers (i.e. on ticket creation and/or ticket update) and then get updated on the telegram about relevant tickets.

**Comment by amigthea** [comment]
> Any update on this issue?
> 
> I see there is a telegram bot integration implemented so should be very easy to implement this feature to notify about new / pending / etc tickets similar as there is a Slack notification implemented.
> 
> EDIT:
> 
> For those who are looking to resolve this: google for zammad triggers - https://admin-docs.zammad.org/en/latest/manage/trigger.html
> 
> Basically, you create a trigger with custom hook notification, paste there an URL with your script.
> 
> Script needs to handle `Content-type: application/json` as zammad will send this content with POST method to your URL.
> 
> The script should then handle the info and post to the telegram.
> 
> You can also create multiple triggers (i.e. on ticket creation and/or ticket update) and then get updated on the telegram about relevant tickets.

can you please instruct me on how I can create the script? or some source where I can learn to do that?

**Comment by MrGeneration** [comment]
Please don't use issues for technical questions or bump them. 
Locking conversation to contributors only.

Please use the community if you need assistance.

**Comment by dominikklein** [comment]
At least it would be possible with the next release to send some notification messages inside a chat/group with the new custom payload possibility in webhooks (https://github.com/zammad/zammad/issues/4483).

_Something in this direction:_

Endpoint: `https://api.telegram.org/bot<token>/sendMessage`
Custom-Payload:
```
{
  "chat_id": "-713149919",
  "text": "#{ticket.title}\n\n[Ticket##{ticket.number}](#{notification.link}): #{notification.message}",
  "parse_mode": "markdown",
  "disable_notification": true
}
```

**Comment by MrGeneration** [comment]
Bots cannot start initial interactions and require the user to type `/start`  in the bot chat. At least as far as I'm aware.

**Comment by dominikklein** [comment]
Yes, it's more a general notification inside an existing channel, I think something else is not really possible.
